src/Voron/VoronExceptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Voron.Impl;
+
+namespace Voron
+{
+    public static class VoronExceptions
+    {
+        public static void ThrowIfReadOnly(LowLevelTransaction tx, string message = "A write transaction is required for this operation")
```

Comment:
Why do we no longer apply `[DoesNotReturn]` to methods like that?

Reply:
Because in this case they are throw if, there is still an option for it not to fail. I will check later if I still need to add a `Throw` method marked with `[DoesNotReturn]`.